src/actions/smartWalletActions.js

```diff
@@ -1192,14 +1198,19 @@ export const settleTransactionsAction = (txToSettle: TxToSettle[]) => {
       });
 
     if (txHash) {
-      const { accounts: { data: accounts } } = getState();
+      const {
+        accounts: { data: accounts },
+        assets: { supportedAssets },
+      } = getState();
+      const accountAssets = accountAssetsSelector(getState());
+      const accountAssetsData = getAssetsAsList(accountAssets);
       const accountId = getActiveAccountId(accounts);
       const accountAddress = getActiveAccountAddress(accounts);
-      const settlementData = txToSettle.map(({ symbol, value, hash }) => ({
-        symbol,
-        value: value.toString(),
-        hash,
-      }));
+      const settlementData = txToSettle.map(({ symbol, value, hash }) => {
+        const { decimals = 18 } = getAssetData(accountAssetsData, supportedAssets, symbol);
+        const parsedValue = parseFloat(value) * (10 ** decimals);
```

Comment:
we need to be sure that we store that number as integer (or string), but not the float